reactioncommerce/reaction
# (refactor) use Events.Hooks instead of meteor collection hooks for cart events that trigger discount calculations

Resolves: [3635](https://github.com/reactioncommerce/reaction/issues/3635)
Impact: Minor
Type: Refactor

## Issue
The cart uses hooks from `meteor-collection-hooks` package, this package will be removed.

## Solution
Convert cart collection hooks to use Reaction's [Hooks.Events](https://blog.reactioncommerce.com/reaction-architecture-events/). 

## Breaking changes
none

## Testing Instructions
Discounts
- Set a discount code and discount amount, a value of 10 would translate to 10% off
- Add a product to the cart
- checkout and apply the discount code
- Finish checkout and verify order confirmation email shows the discount.

Adding items to cart
- Add an item to the cart
- Verify the corresponding inventory document has its workflow field set to `reserved`

Removing item from cart
- Add item to cart 
- Remove item from cart
- Verify the corresponding inventory document has its workflow field set to `new`
